lib/route.js

```diff
@@ -154,43 +155,61 @@ function buildRouting (options) {
 
     const prefix = this[kRoutePrefix]
 
-    this.after((notHandledErr, done) => {
-      const path = opts.url || opts.path
-      if (path === '/' && prefix.length > 0 && opts.method !== 'HEAD') {
-        switch (opts.prefixTrailingSlash) {
-          case 'slash':
-            afterRouteAdded.call(this, { path }, notHandledErr, done)
-            break
-          case 'no-slash':
-            afterRouteAdded.call(this, { path: '' }, notHandledErr, done)
-            break
-          case 'both':
-          default:
-            afterRouteAdded.call(this, { path: '' }, notHandledErr, done)
-            // If ignoreTrailingSlash is set to true we need to add only the '' route to prevent adding an incomplete one.
-            if (ignoreTrailingSlash !== true) {
-              afterRouteAdded.call(this, { path, prefixing: true }, notHandledErr, done)
-            }
-        }
-      } else if (path[0] === '/' && prefix.endsWith('/')) {
-        // Ensure that '/prefix/' + '/route' gets registered as '/prefix/route'
-        afterRouteAdded.call(this, { path: path.slice(1) }, notHandledErr, done)
-      } else {
-        afterRouteAdded.call(this, { path }, notHandledErr, done)
+    const path = opts.url || opts.path
+    if (path === '/' && prefix.length > 0 && opts.method !== 'HEAD') {
+      switch (opts.prefixTrailingSlash) {
+        case 'slash':
+          addNewRoute.call(this, path)
+          break
+        case 'no-slash':
+          addNewRoute.call(this, '')
+          break
+        case 'both':
+        default:
+          addNewRoute.call(this, '')
+          // If ignoreTrailingSlash is set to true we need to add only the '' route to prevent adding an incomplete one.
+          if (ignoreTrailingSlash !== true) {
+            addNewRoute.call(this, path, true)
+          }
       }
-    })
+    } else if (path[0] === '/' && prefix.endsWith('/')) {
+      // Ensure that '/prefix/' + '/route' gets registered as '/prefix/route'
+      addNewRoute.call(this, path.slice(1))
+    } else {
+      addNewRoute.call(this, path)
+    }
+
+    // TODO: remove once review is done
```

Comment:
I'm just preserving these comments as a base for doing further changes if needed based on the old implementation. Once everything approved, I'll delete them

Reply:
Can we remove this now?